packages/subgraph/src/mapping.ts

```diff
@@ -197,20 +197,25 @@ export function handleSetContinuousSwap(event: LOG_SET_FLOW): void {
   pooledInToken.volume = pooledInToken.volume.plus(amountInSinceLastSet);
   pooledInToken.save();
 
-  swap.timestamp = event.block.timestamp;
-  swap.transaction = event.transaction.hash.toHex();
-  swap.minOut = convertTokenToDecimal(event.params.minOut, tokenOut.decimals);
-  swap.maxOut = convertTokenToDecimal(event.params.maxOut, tokenOut.decimals);
-  swap.rateIn = convertTokenToDecimal(event.params.tokenRateIn, tokenIn.decimals);
-  swap.save();
-
   tokenIn.instantSwapCount = tokenIn.instantSwapCount.plus(ONE_BI);
   tokenOut.instantSwapCount = tokenOut.instantSwapCount.plus(ONE_BI);
   pool.instantSwapCount = pool.instantSwapCount.plus(ONE_BI);
   tokenIn.save();
   tokenOut.save();
   pool.save();
 
+  if(event.params.tokenRateIn.equals(ZERO_BI)) {
+    store.remove('ContinuousSwap', swap.id);
```

Comment:
Why would you delete this? It should just be marked as inactive and serve as a historical record.

Reply:
2 reasons:

1. graph preserves all historical record by using the `block` property in the query. therefore it is unnecessary to keep anything historical
2. web ui wants to a list of the user's _current_ swaps. it does that by simply running a user query, and receiving the continuousSwap subdocuments. the query becomes more complicated if we start filtering by activity or other factors.

Reply:
1. then why don't we delete instant swaps as well?
2. that is why you set active to false so you can filter.

Reply:
At a minimum, if you are going to change the domain this way, you need to actually change the grapql schema to reflect your decisions.